modules/mobile/app/controllers/mobile/v1/immunizations_controller.rb

```diff
@@ -29,9 +27,30 @@ def service
       def pagination_params
         Mobile::V0::Contracts::GetPaginatedList.new.call(
           page_number: params.dig(:page, :number),
-          page_size: params.dig(:page, :size)
+          page_size: params.dig(:page, :size),
+          use_cache: use_cache?
         )
       end
+
+      def immunizations
+        immunizations = Mobile::V0::Immunization.get_cached(@current_user) if use_cache?
+
+        if immunizations
+          Rails.logger.info('mobile immunizations cache fetch', user_uuid: @current_user.uuid)
+        else
+          immunizations = immunizations_adapter.parse(service.get_immunizations)
+          Mobile::V0::Immunization.set_cached(@current_user, immunizations)
+          Rails.logger.info('mobile immunizations service fetch', user_uuid: @current_user.uuid)
+        end
+
+        immunizations.reverse
+      end
+
+      def use_cache?
```

Comment:
It wasn't clear if we should add this param or not, but we do use it elsewhere and I thought it could be helpful for testing purposes to be able to bust the cache. I'm definitely open to removing the param.

Reply:
I think adding it is good for the purpose you mentioned.